webroot/src/models/User/User.model.ts

```diff
@@ -331,3 +336,19 @@ export class UserSerializer {
         return new User(userData);
     }
 }
+
+export class LicenseeUserSerializer {
```

Comment:
Makes sense. To complete the separation, would it make sense to also split the models, and possibly also the stores?

Reply:
Probably. But for that one I feel like there are both pros and cons:

Pros:
- No confusion about which type of user we are looking at
- No need to nest into / check licensee param
- No potential to think the permissions param will be populated when it wont

Cons:

- When developing user UIs it might be nice to be able to refer to the common fields post-serialization
- Common getters and helper methods based on common fields post serialization would be repeated. That being said some of the the helper methods like (permissionsShortDisplay) cant be used on the licensee user and you have to just know that atm.


So overall, probably would be worth it but its not as clear cut imo because there is overlap post serialization

Reply:
Im not sure I've seen us do this in a model before, but I've seen times that it may make sense. 

Would we maybe want a user class that has these common model things and then have the staffUser and licenseeUser extend that class and add the things specific to that type of user?

Reply:
Looking into it more there is going to be alot repeated in the actions, state and mutations file so, Im pretty ambivalent to this.

Im thinking the best thing might be to make a new model and alter the store stuff to work with the new model but not have two separate store instances. I think that would capture essentially only the good parts and not the bad parts